### ISSUE TITLE
consider-using-enumerate doesn't trigger if iterated object is an attribute

### ISSUE THREAD

**Issue by natedogith1** [PyCQA/pylint#3657]
consider-using-enumerate isn't triggered if the iterated object is more complex than just a variable name.  (for instance, if it's a property on an object)

pylint 2.4.4
astroid 2.3.3
Python 3.7.3+ (heads/3.7:e64d21b187, Apr 23 2019, 10:57:08)

```python
def foo(arg):
    for i in range(len(arg.bar)): # fails to trigger consider-using-enumerate
        b = arg.bar[i]
```
